qt/python/mantidqt/plotting/markers.py

```diff
@@ -786,25 +821,35 @@ def add_annotate(self, text):
         if text is None:
             return
 
+        if not self.label_visible:
+            self.annotations[self.name] = ''
+            return
+
         marker_in_scope = True
         x_lower, x_upper = self.marker.axis.get_xlim()
         y_lower, y_upper = self.marker.axis.get_ylim()
         if self.marker_type == 'YSingle':
+            x_pos = self.label_x_pos
+            y_pos = self.relative(self.marker.y, y_lower, y_upper) + self.label_y_pos
```

Comment:
it took me a while, but is `self.label_y_pos` actually the offset from the y position?